tests/test_backend_failure.py

```diff
@@ -201,6 +201,24 @@ def test_backend_failure_sanity_customrepo_update_redirect(aktualizr, uptane_rep
     install_result = director.wait_for_install()
     return install_result and update_hash == aktualizr.get_current_image_info(aktualizr.id)
 
+"""
+    Verifies if aktualizr rejects redirects over 10 times - update fails after redirect
+"""
+@with_uptane_backend(start_generic_server=True)
+@with_customrepo(handlers=[
+                            RedirectHandler(number_of_redirects=11, url='/primary-image.img')
+                        ])
+@with_imagerepo()
+@with_director()
+@with_aktualizr(run_mode='once', output_logs=True)
+def test_backend_failure_sanity_customrepo_unsuccessful_update_redirect(aktualizr, uptane_repo,
+                                                           custom_repo, director, **kwargs):
+    update_hash = uptane_repo.add_image(aktualizr.id, 'primary-image.img',
+                                        custom_url=custom_repo.base_url + '/' + 'primary-image.img')
+    install_result = director.wait_for_install()
+    return not install_result
+
+
```

Comment:
The test should look like this

`"""
    Verifies if aktualizr rejects redirects over 10 times - update fails after redirect
"""
@with_uptane_backend(start_generic_server=True)
@with_customrepo(handlers=[
                            RedirectHandler(number_of_redirects=(11 * 3 + 1), url='/primary-image.img')
                        ])
@with_imagerepo()
@with_director()
@with_aktualizr(start=False, run_mode='once', output_logs=True)
def test_backend_failure_sanity_customrepo_unsuccessful_update_redirect(aktualizr, uptane_repo,
                                                           custom_repo, director, **kwargs):
    update_hash = uptane_repo.add_image(aktualizr.id, 'primary-image.img',
                                        custom_url=custom_repo.base_url + '/' + 'primary-image.img')
    with aktualizr:
        aktualizr.wait_for_completion()

    return not director.get_install_result()`

A number of requests should be equal to (CURL_MAX_REDIRECTS + 1) * (RETRY_NUMB) + 1, which is in this specific case is (10+1)*3 + 1 (An initial request to the mock repo server is not considered as a redirect request by curl so it starts counting only from the second request, hence +1).

Reply:
Thanks, @mike-sul . I'll have a try.

Reply:
@mike-sul , the reason of using `aktualizr.wait_for_completion()` instand of `director.wait_for_install()` is "limitation of a redirect number should be aktualizr-wide and applied to each outgoing request.", right?

Reply:
This is a negative test, an installation of a new update is not supposed to happen since a number of redirects exceeds the maxim allowed. Therefore, director.wait_for_install() will never succeed (installation should not happen) and exit with a timeout error. So, the test just simply wait until Aktualizr completes an uptane cycle.